src/plot_api/plot_schema.js

```diff
@@ -437,6 +437,8 @@ function recurseIntoValObject(valObject, parts, i) {
     return valObject;
 }
 
+// note: this is different from Lib.isIndex, this one doesn't accept numeric
+// strings, only actual numbers.
```

Comment:
I initially made `isIndex` into a `Lib` function thinking I could reuse it here... but in this context it's in principle important that an index is actually a number, to distinguish an array index from an object property that happens to be a number. That said I don't think it's possible to use a number as an object property in `nestedProperty`...

Reply:
Oh we could use this for validating `selectedpoints` items as done here:

https://github.com/plotly/plotly.js/blob/a132b85b899dd490216d12823be56ad3861f091f/src/lib/index.js#L497-L499

Reply:
... by _this_ I mean `Lib.isIndex`

Reply:
Good call -> https://github.com/plotly/plotly.js/pull/2629/commits/cdcd6200907d5caad164933ebbf388bc6532bf19

We probably either have or *should* have this in other places too... `mesh3d` is the other obvious place, but it has bigger issues -> #2630